modules/widgets/blog-stats.php

```diff
@@ -0,0 +1,177 @@
+		}
+	}
+
+	/**
+	 * Back-end widget form.
+	 *
+	 * @see WP_Widget::form()
+	 *
+	 * @param array $instance Previously saved values from database.
+	 *
+	 * @return void
+	 */
+	function form( $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults() );
+		?>
+
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>"><?php esc_html_e( 'Title:', 'jetpack' ); ?></label>
+			<input class="widefat" id="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'title' ) ); ?>" type="text" value="<?php echo esc_attr( $instance['title'] ); ?>" />
+		</p>
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'hits' ) ); ?>"><?php echo number_format_i18n( '12345' ); ?></label>
+			<input class="widefat" id="<?php echo esc_attr( $this->get_field_id( 'hits' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'hits' ) ); ?>" type="text" value="<?php echo esc_attr( $instance['hits'] ); ?>" />
+		</p>
+		<p><?php esc_html_e( 'Hit counter is delayed by up to 60 seconds.', 'jetpack' ); ?></p>
+
+		<?php
+	}
+
+	/**
+	 * Sanitize widget form values as they are saved.
+	 *
+	 * @see WP_Widget::update()
+	 *
+	 * @param array $new_instance Values just sent to be saved.
+	 * @param array $old_instance Previously saved values from database.
+	 *
+	 * @return array Updated safe values to be saved.
+	 */
+	function update( $new_instance, $old_instance ) {
+		$instance          = array();
+		$instance['title'] = wp_kses( $new_instance['title'], array() );
+		$instance['hits']  = wp_kses( $new_instance['hits'], array() );
+
+		return $instance;
+	}
+
+	/**
+	 * Front-end display of widget.
+	 *
+	 * @see WP_Widget::widget()
+	 *
+	 * @param array $args     Widget arguments.
+	 * @param array $instance Saved values from database.
+	 */
+	function widget( $args, $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults() );
+
+		/** This filter is documented in wp-includes/widgets/class-wp-widget-pages.php */
+		$title = apply_filters( 'widget_title', $instance['title'] );
+
+		echo $args['before_widget'];
+
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . esc_html( $title ) . $args['after_title'];
+		}
+
+		// Get the Site Stats.
+		$views = $this->get_stats();
+
+		if ( isset( $views ) && ! empty( $views ) ) {
+			printf(
+				'<ul><li>%1$s %2$s</li></ul>',
+				number_format_i18n( $views ),
+				isset( $instance['hits'] ) ? esc_html( $instance['hits'] ) : ''
+			);
+		} else {
+			esc_html_e( 'No hits.', 'jetpack' );
+		}
+
+		echo $args['after_widget'];
+
+		/** This action is already documented in modules/widgets/gravatar-profile.php */
+		do_action( 'jetpack_stats_extra', 'widget_view', 'blog_stats' );
+	}
+}
+
+/**
+ * If the Stats module is active in a recent version of Jetpack, register the widget.
+ *
+ * @since 4.5.0
+ */
+function jetpack_blog_stats_widget_init() {
+	if ( ! function_exists( 'stats_get_from_restapi' ) ) {
```

Comment:
method doesn't actually return something if function exists so it would be better to do only:
```php
if ( function_exists( 'stats_get_from_restapi' ) ) {
    register_widget( 'Jetpack_Blog_Stats_Widget' );
}
```
we also save the `!` inversion, that's at least 1 millisecond :D